File: frontend/src/api/comment.ts
import { http } from './request'

// 评论信息
export interface CommentInfo {
  id: number
  postId: number
  userId: number
  parentId: number
  content: string
  isEdited: number
  likeCount: number
  replyCount: number
  status: number // 0-待审核 1-正常 2-已删除
  createAt: string
  updateAt: string
  replies?: CommentInfo[] // 楼中楼回复
}

// 创建评论请求
export interface CreateCommentRequest {
  postId: number
  parentId?: number
  content: string
}

// 更新评论请求
export interface UpdateCommentRequest {
  id: number
  content: string
}

// 评论列表响应
export interface CommentListResponse {
  totalCount: number
  hasMore: boolean
  comments: CommentInfo[]
}

// 评论 API
export const commentApi = {
  // 创建评论
  create(data: CreateCommentRequest): Promise<CommentInfo> {
    return http.post('/comment', data)
  },

  // 获取评论详情
  getById(id: number): Promise<CommentInfo> {
    return http.get(`/comment/${id}`)
  },

  // 获取文章评论列表（顶级评论）
  getByPostId(postId: number, params?: { page?: number; pageSize?: number }): Promise<CommentListResponse> {
    return http.get(`/comment/post/${postId}`, { params })
  },

  // 获取楼中楼回复列表
  getReplies(parentId: number, params?: { page?: number; pageSize?: number }): Promise<CommentListResponse> {
    return http.get(`/comment/reply/${parentId}`, { params })
  },

  // 更新评论
  update(id: number, data: UpdateCommentRequest): Promise<CommentInfo> {
    return http.put(`/comment/${id}`, data)
  },

  // 删除评论
  delete(id: number): Promise<void> {
    return http.delete(`/comment/${id}`)
  },

  // 点赞评论
  like(id: number): Promise<void> {
    return http.post(`/comment/${id}/like`)
  },

  // 取消点赞评论
  unlike(id: number): Promise<void> {
    return http.post(`/comment/${id}/unlike`)
  },
}

File: internal/apiserver/model/comment.go
package model

import "time"

// Comment 评论模型
type Comment struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	PostID      uint      `gorm:"column:post_id;type:bigint unsigned;not null;index:idx_comments_post_id;comment:文章 ID" json:"postId"`
	UserID      uint      `gorm:"column:user_id;type:bigint unsigned;not null;index:idx_comments_user_id;comment:评论者 ID" json:"userId"`
	ParentID    uint      `gorm:"column:parent_id;type:bigint unsigned;default:0;index:idx_comments_parent_id;comment:父评论 ID（楼中楼）" json:"parentId"`
	Content     string    `gorm:"column:content;type:text;not null;comment:评论内容" json:"content"`
	IsEdited    int       `gorm:"column:is_edited;type:tinyint unsigned;not null;default:0;comment:是否已编辑：0-否 1-是" json:"isEdited"`
	LikeCount   uint      `gorm:"column:like_count;type:int unsigned;not null;default:0;comment:点赞数" json:"likeCount"`
	ReplyCount  uint      `gorm:"column:reply_count;type:int unsigned;not null;default:0;comment:回复数" json:"replyCount"`
	Status      int       `gorm:"column:status;type:tinyint unsigned;not null;default:1;comment:状态：0-待审核 1-正常 2-已删除" json:"status"`
	CreateAt    time.Time `gorm:"column:created_at;type:datetime;not null;default:CURRENT_TIMESTAMP;comment:创建时间" json:"createAt"`
	UpdateAt    time.Time `gorm:"column:updated_at;type:datetime;not null;default:CURRENT_TIMESTAMP;comment:更新时间" json:"updateAt"`
}

// TableName 表名
func (Comment) TableName() string { return "comments" }

// 创建评论请求结构
type CreateCommentRequest struct {
	PostID   uint   `json:"postId" binding:"required"`
	ParentID uint   `json:"parentId" binding:"omitempty"`
	Content  string `json:"content" binding:"required,max=1000"`
}

// 更新评论请求结构
type UpdateCommentRequest struct {
	ID      uint   `json:"id" binding:"required"`
	Content string `json:"content" binding:"required,max=1000"`
}

// 评论响应结构
type CommentInfo struct {
	ID         uint      `json:"id"`
	PostID     uint      `json:"postId"`
	UserID     uint      `json:"userId"`
	ParentID   uint      `json:"parentId"`
	Content    string    `json:"content"`
	IsEdited   int       `json:"isEdited"`
	LikeCount  uint      `json:"likeCount"`
	ReplyCount uint      `json:"replyCount"`
	Status     int       `json:"status"`
	CreateAt   string    `json:"createAt"`
	UpdateAt   string    `json:"updateAt"`
}

// 评论列表响应
type CommentListResponse struct {
	TotalCount int64         `json:"totalCount"`
	HasMore    bool          `json:"hasMore"`
	Comments   []CommentInfo `json:"comments"`
}
